<launch>
  <arg name="launch_main" default="true" />
  <arg name="json" />

  <node pkg="jsk_2016_01_baxter_apc" type="stow_work_order_server.py" name="strategic_work_order" respawn="true" output="screen" clear_params="true">
    <rosparam subst_value="true">
        json: $(arg json)
        limit_volume: 5000
        volume_first:
          - kleenex_paper_towels
          - dr_browns_bottle_brush
          - hanes_tube_socks
        black_list: []
    </rosparam>
  </node>

  <!-- parameter -->
  <param name="/left_hand/state" value="" />
  <param name="/left_hand/target_bin" value="" />
  <param name="/left_hand/target_object" value="" />
  <param name="/left_hand/update_json" value="false" />
  <param name="/right_hand/state" value="" />
  <param name="/right_hand/target_bin" value="" />
  <param name="/right_hand/target_object" value="" />
  <param name="/right_hand/update_json" value="false" />

  <node name="yes_no_button"
        pkg="jsk_2016_01_baxter_apc" type="yes_no_button" >
  </node>

  <node pkg="jsk_2016_01_baxter_apc" type="output_json_stow.py" name="output_json_stow" respawn="true">
    <rosparam subst_value="true">
        json: $(arg json)
    </rosparam>
  </node>

  <!-- object verification -->
  <group ns="left_hand">
    <node name="object_verification"
          pkg="jsk_2016_01_baxter_apc" type="apply_tote_contents_hint.py"
          respawn="true">
      <remap from="~input" to="vgg16_object_recognition/output" />
      <param name="json" value="$(arg json)" />
    </node>
  </group>
  <group ns="right_hand">
    <node name="object_verification"
          pkg="jsk_2016_01_baxter_apc" type="apply_tote_contents_hint.py"
          respawn="true">
      <remap from="~input" to="vgg16_object_recognition/output" />
      <param name="json" value="$(arg json)" />
    </node>
  </group>

  <!-- robot main -->
  <node if="$(arg launch_main)" pkg="jsk_2016_01_baxter_apc" type="main-stow.l" name="right_hand"
    args="&quot;(progn (apc-stow-init :rarm-controller) (apc-stow-mainloop :rarm))&quot;"
    respawn="true" output="screen">
    <env name="DISPLAY" value="" />
  </node>
  <node if="$(arg launch_main)" pkg="jsk_2016_01_baxter_apc" type="main-stow.l" name="left_hand"
    args="&quot;(progn (apc-stow-init :larm-controller) (apc-stow-mainloop :larm))&quot;"
    respawn="true" output="screen">
    <env name="DISPLAY" value="" />
  </node> 

  <!-- visualization -->
  <node name="visualize_json"
        pkg="jsk_apc2016_common" type="visualize_stow_json.py"
        args="$(arg json)" />
  <!--
  <node name="visualize_on_rviz"
        pkg="jsk_2015_05_baxter_apc" type="visualize_on_rviz.py">
    <param name="json" value="$(arg json)" />
  </node>
  -->
  <node name="rosconsole_overlay_text"
        pkg="jsk_rviz_plugins" type="rosconsole_overlay_text.py">
    <rosparam>
      nodes:
        - /left_hand
        - /right_hand
    </rosparam>
  </node>

  <!-- set tote_contents rosparam -->
  <node name="set_tote_contents_param"
        pkg="jsk_2016_01_baxter_apc" type="json_to_rosparam.py">
    <remap from="~param" to="/tote_contents" />
    <rosparam subst_value="true">
      json: $(arg json)
      key: tote_contents
    </rosparam>
  </node>

</launch>
